mokapot/model.py

```diff
@@ -276,12 +280,33 @@ def fit(self, psms):
                 psms = copy.copy(psms)
                 psms._data = psms._data.iloc[subset_idx, :]
 
-        # Choose the initial direction
-        start_labels, feat_pass = _get_starting_labels(psms, self)
-
         # Normalize Features
-        self.features = psms.features.columns.tolist()
         norm_feat = self.scaler.fit_transform(psms.features.values)
+        self.features = psms.features.columns.tolist()
+
+        # Supervised step
+        if self.supervised_step:
+            # Shuffle input order
+            size = psms.__len__()
+            my_shuffled_idx = np.random.permutation(np.arange(size))
+            my_original_idx = np.argsort(my_shuffled_idx)
+            shuffled_norm_feat = norm_feat[my_shuffled_idx, :]
+            shuffled_targets = psms.targets[my_shuffled_idx]
```

Comment:
We shouldn't always need to shuffle, and there is already shuffling code below. Remove this lines.